presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcStorageManager.java

```diff
@@ -188,6 +189,31 @@ public void shutdown()
         deletionExecutor.shutdownNow();
     }
 
+    private static TypeSignature toStorageType(TypeSignature typeSignature)
+    {
+        List<TypeSignatureParameter> typeSignatureParameterList = typeSignature.getParameters();
+        switch (typeSignature.getBase()) {
+            case StandardTypes.TIMESTAMP:
+            case StandardTypes.DATE:
+                return BIGINT.getTypeSignature();
+            case StandardTypes.ARRAY:
+                checkArgument(typeSignatureParameterList.size() == 1, "ARRAY type must have one element");
+                return new TypeSignature(StandardTypes.ARRAY, ImmutableList.of(TypeSignatureParameter.of(toStorageType(typeSignatureParameterList.get(0).getTypeSignature()))));
+            case StandardTypes.MAP:
+                checkArgument(typeSignatureParameterList.size() == 2, "MAP type must have two elements");
+                return new TypeSignature(StandardTypes.MAP, ImmutableList.of(
+                        TypeSignatureParameter.of(toStorageType(typeSignatureParameterList.get(0).getTypeSignature())),
+                        TypeSignatureParameter.of(toStorageType(typeSignatureParameterList.get(1).getTypeSignature()))));
+            case StandardTypes.ROW:
+                ImmutableList.Builder<TypeSignatureParameter> builder = ImmutableList.builder();
+                for (TypeSignatureParameter typeSignatureParameter : typeSignatureParameterList) {
+                    builder.add(TypeSignatureParameter.of(toStorageType(typeSignatureParameter.getTypeSignatureOrNamedTypeSignature().get())));
```

Comment:
I don't see `getTypeSignatureOrNamedTypeSignature()` used outside of `TypeRegistry`. Why are you using it here?